onset_maproom/maproom.py

```diff
@@ -82,6 +84,45 @@ def map_click(click_lat_lng):
         children=dlf.Tooltip("({:.3f}, {:.3f})".format(*lat_lng))
     )
 
+@APP.callback(
+    Output("plotly_onset_test", "figure"),
+    Input("map", "click_lat_lng"),
+    Input("earlyStartDay", "value"),
+    Input("earlyStartMonth", "value"),
+    Input("searchDays", "value"),
+    Input("wetThreshold", "value"),
+    Input("runningDays", "value"),
+    Input("runningTotal", "value"),
+    Input("minRainyDays", "value"),
+    Input("dryDays", "value"),
+    Input("drySpell", "value"),
+)
+def onset_plot(click_lat_lng, earlyStartDay, earlyStartMonth, searchDays, wetThreshold, runningDays, runningTotal, minRainyDays, dryDays,drySpell):
+    lat, lng = get_coords(click_lat_lng)
+    ds = rr_mrg.sel(X=lng, Y=lat, method="nearest", drop=True)
+    onsetDays = calc.onset_date(ds.precip, int(earlyStartDay), \
+        calc.strftimeb2int(earlyStartMonth), searchDays, \
+        wetThreshold, runningDays, runningTotal, minRainyDays, dryDays, drySpell)
+    onsetDate = onsetDays["T"] + onsetDays
+    year = pd.DatetimeIndex(onsetDate['T']).year
+    onsetMD = onsetDate.dt.strftime("2000-%m-%d").to_dataframe(name="Onset Date")
+    onsetMD['Year'] = year
+    graph = px.line(
+        data_frame=onsetMD,
+        x="Year", 
+        y="Onset Date",
+    )
+    graph.update_traces(
+        mode="markers+lines",
+        hovertemplate='%{y} %{x}'
+    )
+    graph.update_layout(
+        yaxis=dict(tickformat="%b %d"), 
+        xaxis_title="Year", 
+        yaxis_title="Onset Date",
+        title= f"Starting dates of {int(earlyStartDay)} {earlyStartMonth} season {year.min()}-{year.max()} ({lat}E, {lng}N)"
+    )
```

Comment:
Which lat/lon is being printed? I would print the result of nearest point, as it is what the data is. Thus, given the data spatial resolution is 0.0375, I would make sure that it prints at 4 digits after decimal point so that there is not ambiguity which data spatial point is being looked at (currently only 2 digits).

Reply:
this final 1 is just a machine error reading/writing. Here 4 as we know the resolution is 0.0375